content/courses/advanced-ada/parts/data_types/numerics.rst

```diff
@@ -1329,22 +1329,457 @@ input parameter :ada:`V` (of :ada:`One_To_Ten'Base` type) in the runtime output
 of the program before it's adapted to fit the constraints of the
 :ada:`One_To_Ten` subtype.
 
-.. ::
 
-    .. _Adv_Ada_Discrete_Numeric_Types:
+.. _Adv_Ada_Discrete_Numeric_Types:
 
-    Discrete Numeric Types
-    ----------------------
+Discrete Numeric Types
+----------------------
 
-    .. todo::
+In the Introduction to Ada course, we've seen that Ada has two kinds of integer
+type: :ref:`signed integer <Intro_Ada_Integers>` and
+:ref:`modular <Intro_Ada_Unsigned_Types>` types. For example:
 
-        Complete section!
+.. code:: ada compile_button project=Courses.Advanced_Ada.Data_Types.Numerics.Discrete_Numeric_Types.Signed_Modular_Types
+
+    package Num_Types is
+
+       type Signed_Integer is range 1 .. 1_000_000;
+       type Modular is mod 2**32;
+
+    end Num_Types;
+
+Remember that modular types are similar to *unsigned* integer types in other
+programming languages.
+
+In this section, we review these types and look into a couple of details that
+haven't been covered yet. We start the discussion with
+:ref:`signed integer types <Adv_Ada_Integer_Types>`, and then move on to
+:ref:`modular types <Adv_Ada_Modular_Types>`.
+
+.. _Adv_Ada_Integer_Types:
+
+Integer types
+~~~~~~~~~~~~~
+
+In the :ref:`Introduction to Ada <Intro_Ada_Integers>` course, we mentioned
+that you can define your own integer types in Ada. In fact, you're almost
+expected to do so, as Ada only guarantees the existence of a single integer
+type |mdash| and a few optional integer types. Even though a specific compiler
```

Comment:
Maybe "type |mdash| and a few optional integer types." should be
"type |mdash| and the names of a few optional integer types."  ??